static_frame/core/index_hierarchy.py

```diff
@@ -1485,26 +2170,32 @@ def isin(self, other: tp.Iterable[tp.Iterable[tp.Hashable]]) -> np.ndarray:
 
         return isin(self.flat().values, matches)
 
-    def roll(self, shift: int) -> 'IndexHierarchy':
-        '''Return an :obj:`IndexHierarchy` with values rotated forward and wrapped around (with a positive shift) or backward and wrapped around (with a negative shift).
+    def roll(self: IH,
+            shift: int,
+            ) -> IH:
+        '''
+        Return an :obj:`IndexHierarchy` with values rotated forward and wrapped around (with a positive shift) or backward and wrapped around (with a negative shift).
         '''
         if self._recache:
-            self._update_array_cache()
+            self._update_array_cache() # pragma: no cover
```

Comment:
Why skip coverage of this scenario?

Reply:
See this response: https://github.com/InvestmentSystems/static-frame/pull/437#discussion_r831501537